modules/provision/engine.go

```diff
@@ -116,16 +129,39 @@ func (e *defaultEngine) decommission(ctx context.Context, r *Reservation) error
 	return nil
 }
 
-func (e *defaultEngine) reply(id string, result interface{}, err error) {
-	//TODO: actually push the reply to the endpoint defined by `to`
-	if err != nil {
+func (e *defaultEngine) reply(ctx context.Context, r *Reservation, rErr error, info interface{}) error {
+	zbus := GetZBus(ctx)
+	identity := stubs.NewIdentityManagerStub(zbus)
+	result := &Result{}
```

Comment:
why u do this syntax instead of 
```suggestion
	var result Result
```

Reply:
What you suggest doesn't produce the same result btw.
I'm creating a pointer to a Result object,
You are creating an Result object

regarding the syntax this is just by habit. I always user literal whenever I can.